native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/tasks/GenerateDynamicAccessMetadata.java

```diff
@@ -0,0 +1,202 @@
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.graalvm.buildtools.gradle.tasks;
+
+import com.github.openjson.JSONArray;
+import com.github.openjson.JSONObject;
+import org.graalvm.buildtools.gradle.internal.GraalVMLogger;
+import org.graalvm.buildtools.gradle.internal.GraalVMReachabilityMetadataService;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Generates a {@code dynamic-access-metadata.json} file used by the dynamic access tab of the native image
+ * Build Report. This json file contains the mapping of all classpath entries that exist in the
+ * {@value #LIBRARY_AND_FRAMEWORK_LIST} to their transitive dependencies.
+ * <p>
+ * If {@value #LIBRARY_AND_FRAMEWORK_LIST} doesn't exist in the used release of the
+ * {@code GraalVM Reachability Metadata} repository, this task does nothing.
+ * <p>
+ * The format of the generated JSON file conforms the following
+ * <a href="https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/assets/dynamic-access-metadata-schema-v1.0.0.json">schema</a>.
+ */
+public abstract class GenerateDynamicAccessMetadata extends DefaultTask {
+    private static final String LIBRARY_AND_FRAMEWORK_LIST = "library-and-framework-list.json";
+    private static final String ARTIFACT = "artifact";
+    private static final String METADATA_PROVIDER = "metadataProvider";
+    private static final String PROVIDES_FOR = "providesFor";
+
+    @Internal
+    public abstract Property<Configuration> getRuntimeClasspath();
```

Comment:
:x: This is not compatible with Gradle's configuration cache. We shouldn't use `Configuration` as input (probably a `Property<ResolvedComponentResult>`).